feat: add client-side tools support with onToolCall callback API

omeraplak opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://voltagent.dev/docs/community/contributing/#commit-convention

## Bugs / Features

- [x] Related issue(s) linked
- [x] Tests for the changes have been added
- [x] Docs have been added / updated
- [x] Changesets have been added https://voltagent.dev/docs/community/contributing/#creating-a-changeset

## What is the current behavior?

## What is the new behavior?

fixes (issue)

- #362 

## Notes for reviewers

<!-- Add any notes/questions you may have for reviewers -->

zrosenbauer reviewed (commented):
Love this conceptually but I don't think this actually works on Client no? I would just use vercel by default and get this?

omeraplak commented:
> Love this conceptually but I don't think this actually works on Client no? I would just use vercel by default and get this?

You’re absolutely right! I’ve just updated the PR and added an example using the Vercel AI UI.
Would love to hear what you think. Does it look good to you?
